### ISSUE TITLE
Version Support

### ISSUE THREAD

**Issue by dahall** [dahall/TaskScheduler#201]
Hi,

The current version doesn't support 1.5 versioned tasks.

This is the error we get:
System.InvalidOperationException: The current version of the native library (1.4) does not support the original or minimum version of the ".NET Framework NGEN v4.0.30319" task (1.5/1.5).

will it be supported?

Thank you,

Adam.

Originally posted: 2017-01-26T05:52:17

**Comment by dahall** [comment]
Which version of Windows are you using? It is a pretty easy fix to add another version #, but need some detail on what it may be enabling.

Originally posted: 2017-01-26T06:49:09

**Comment by dahall** [comment]
I've done some research, and even the newest Windows release (Server 2016), hasn't upgraded the Task Scheduler library. It is still version 1.4. Is it possible you have created the task via XML and put 1.5 in the Task element version attribute? If so, it will not work no matter what I do in the wrapper library. The XML for an imported task should look like the following on Windows 10 or Windows Server 2016:
```XML
<?xml version="1.0" encoding="UTF-16"?>
   <Task xmlns="http://schemas.microsoft.com/windows/2004/02/mit/task" version="1.4">
```
For each earlier version of Windows, subtract 0.1 from the version (e.g. Windows 7 == 1.2). The '1.4' you see in the sample XML must be equal to or lower than the number supported by the OS.

Originally posted: 2017-01-27T14:40:04

**Comment by dahall** [comment]
Upon further review, I have identified a bug in the code that will, for Windows 10 systems, cause a version 1.5 to show up. I will correct in release 2.5.22 due shortly.

Originally posted: 2017-01-30T19:09:18

**Comment by dahall** [comment]
Hi,

I tried the newest version Release 2.5.28, changed the dll but the same exception still occur:
when I run through Microsoft.Win32.TaskScheduler.Task results, some results, when checking for result.IsActive will throw the following exception:

"The current version of the native library (1.4) does not support the original or minimum version of the \".NET Framework NGEN v4.0.30319\" task (1.5/1.5)"

Stack Trace:
   at Microsoft.Win32.TaskScheduler.Task.GetV2Definition(TaskService svc, IRegisteredTask iTask, Boolean throwError)
   at Microsoft.Win32.TaskScheduler.Task.get_Definition()
   at Microsoft.Win32.TaskScheduler.Task.get_IsActive()

Example Task:
Name of task : ".NET Framework NGEN v4.0.30319"
Pשאי: "\\Microsoft\\Windows\\.NET Framework\\.NET Framework NGEN v4.0.30319"





Originally posted: 2017-07-23T08:35:17

**Comment by dahall** [comment]
Fixed in 2.5.22

Originally posted: 2017-02-05T13:44:23
